Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp

```diff
@@ -718,9 +763,17 @@ struct RGroupDecompData {
       if (atm == atoms.end()) continue;  // label not used in the rgroup
 
       Atom *atom = atm->second;
-      mappings[indexLabel] = ++count;
+      
```

Comment:
This is the main fix, I wasn't recycling labels, they were always being incremented in the core.  Whups.